RulesEngine/Ruleset.cs

```diff
@@ -102,7 +102,7 @@ public void AddRange(IEnumerable<Rule> collection)
                 {
                     if (_logger != null)
```

Comment:
You don't need this check with the ? on logger below.

Reply:
The ? will, if the thing to the left is null, skip processing whats on the right and return null.  

For example when `_logger` is null.

With `_logger.Debug` you get a null reference error because you're trying to call `Debug` on `null`.

With `_logger?.Debug` instead of calling `Debug` the line just is a no-op.

Reply:
If you have this `if (_logger is null)` pattern elsewhere you can replace it with the one line `?` version.

Reply:
very true...done elsewhere but will make consistent